test/builtin_dataset_mocks.py

```diff
@@ -66,17 +64,17 @@ def prepare(self, home, config):
 
         mock_info = self._parse_mock_info(self.mock_data_fn(root, config))
 
-        with unittest.mock.patch.object(datasets.utils.Dataset2, "__init__"):
-            required_file_names = {
-                resource.file_name for resource in datasets.load(self.name, root=root, **config)._resources()
-            }
-        available_file_names = {path.name for path in root.glob("*")}
-        missing_file_names = required_file_names - available_file_names
-        if missing_file_names:
-            raise pytest.UsageError(
-                f"Dataset '{self.name}' requires the files {sequence_to_str(sorted(missing_file_names))} "
-                f"for {config}, but they were not created by the mock data function."
-            )
+        # with unittest.mock.patch.object(datasets.utils.Dataset2, "__init__"):
+        #     required_file_names = {
+        #         resource.file_name for resource in datasets.load(self.name, root=root, **config)._resources()
+        #     }
+        # available_file_names = {path.name for path in root.glob("*")}
+        # missing_file_names = required_file_names - available_file_names
+        # if missing_file_names:
+        #     raise pytest.UsageError(
+        #         f"Dataset '{self.name}' requires the files {sequence_to_str(sorted(missing_file_names))} "
+        #         f"for {config}, but they were not created by the mock data function."
+        #     )
```

Comment:
Since the resources are now fully internal inside the dataset function, I don't see a good way to check if the mock data is set up correctly. One thing we could try is to patch 

https://github.com/pytorch/vision/blob/d8654bb0d84fd2ba8b42cd58d881523821a6214c/torchvision/prototype/datasets/utils/_resource.py#L117-L119

on all subclasses to raise the error I commented out above.

Reply:
Could `load_images_dp` be registered and mocked?

Reply:
That would require registering functions like that just for testing purposes. Not sure if we should go that way.

Reply:
Then, how about making `info` contains another dictionary of `Resource`s? And, `load_images_dp` could load resource from `info`.

Reply:
That would be a possibility, yes. cc @NicolasHug